src/commands/Minion/craft.ts

```diff
@@ -78,75 +75,59 @@ export default class extends BotCommand {
-				const id = parseInt(itemID);
-				if (id === 995) {
-					const userGP = msg.author.settings.get(UserSettings.GP);
-					if (userGP < qty) {
-						return msg.send('You do not have enough GP.');
-					}
-					quantity = Math.min(quantity, Math.floor(userGP / qty));
-					continue;
-				}
-				const itemsOwned = userBank[parseInt(itemID)];
-				if (itemsOwned === undefined || itemsOwned < qty) {
-					return msg.send(`You dont have enough ${itemNameFromID(parseInt(itemID))}.`);
-				}
-				quantity = Math.min(quantity, Math.floor(itemsOwned / qty));
-			}
+			const max = userBank.fits(craftable.inputItems);
+			if (max < quantity && max !== 0) quantity = max;
 		}
 
 		const duration = quantity * timeToCraftSingleItem;
-
 		if (duration > maxTripLength) {
 			return msg.send(
 				`${msg.author.minionName} can't go on trips longer than ${formatDuration(
 					maxTripLength
-				)}, try a lower quantity. The highest amount of ${Craft.name}s you can craft is ${Math.floor(
+				)}, try a lower quantity. The highest amount of ${craftable.name}s you can craft is ${Math.floor(
 					maxTripLength / timeToCraftSingleItem
 				)}.`
 			);
 		}
 
-		// Check the user has add the required items to craft.
-		for (const [itemID, qty] of requiredItems) {
-			const id = parseInt(itemID);
-			if (id === 995) {
-				const userGP = msg.author.settings.get(UserSettings.GP);
-				if (userGP < qty * quantity) {
-					return msg.send(`You don't have enough ${itemNameFromID(id)}.`);
-				}
-				continue;
-			}
-			if (!bankHasItem(userBank, id, qty * quantity)) {
-				return msg.send(`You don't have enough ${itemNameFromID(id)}.`);
-			}
-		}
+		const itemsNeeded = craftable.inputItems.clone().multiply(quantity);
+		let gpNeeded = 0;
+		if (itemsNeeded.has('Coins')) gpNeeded = itemsNeeded.amount('Coins');
 
-		// Remove the required items from their bank.
-		let newBank = { ...userBank };
-		for (const [itemID, qty] of requiredItems) {
-			if (parseInt(itemID) === 995) {
-				await msg.author.removeGP(qty * quantity);
-				continue;
-			}
-			newBank = removeItemFromBank(newBank, parseInt(itemID), qty * quantity);
+		// Check the user has all the required items to craft.
+		if (!userBank.has(itemsNeeded.bank)) {
+			return msg.send(
+				`You don't have enough items. For ${quantity}x ${craftable.name}, you're missing **${itemsNeeded
+					.clone()
+					.remove(userBank)}**.`
+			);
 		}
-		await msg.author.settings.update(UserSettings.Bank, newBank);
+
+		await msg.author.removeItemsFromBank(itemsNeeded);
+
+		updateBankSetting(this.client, ClientSettings.EconomyStats.CraftingCost, itemsNeeded.bank);
 
 		await addSubTaskToActivityTask<CraftingActivityTaskOptions>({
-			craftableID: Craft.id,
+			craftableID: craftable.id,
 			userID: msg.author.id,
 			channelID: msg.channel.id,
 			quantity,
 			duration,
 			type: Activity.Crafting
 		});
 
+		let gpUsedString = '';
+
+		if (gpNeeded > 0) {
+			gpUsedString += ` and ${gpNeeded} GP`;
+		}
```

Comment:
This part is not needed and causes a weird output

![image](https://user-images.githubusercontent.com/30398469/122971018-e8113700-d3d1-11eb-86a7-2e9ef9ff9346.png)